hvantk/tools/plugins/plugins_cli.py

```
@@ -73,4 +73,36 @@ def validate_cmd(manifest_path: str):
+    catalog_rel = content.get("catalog")
+    if catalog_rel:
+        catalog_path = Path(manifest_path).parent / catalog_rel
+        if not catalog_path.is_file():
+            raise click.ClickException(f"catalog not found: {catalog_path}")
+        try:
+            catalog_schema = json.loads(
+                (Path(plugin_loader.__file__).parent / "catalog_entry.schema.json").read_text()
+            )
+            entries = json.loads(catalog_path.read_text())
+        except (OSError, json.JSONDecodeError) as exc:
+            raise click.ClickException(f"failed to read catalog {catalog_path}: {exc}")
```

Ruff (0.15.15)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)